From 2cb4f3c4f95cbc8a97184673ab951e28f3e17210 Mon Sep 17 00:00:00 2001
From: Sarika Sharma <quic_sarishar@quicinc.com>
Date: Tue, 25 Jun 2024 12:50:39 +0530
Subject: [PATCH] wifi: ath12k: Fix KW issue tx_stats mu_group of size 64 use
 index upto 64

Currently array tx_stats->mu_group size is MAX_MU_GROUP_ID(64).
But while assigning value to array it is iterating from 0 to
MAX_MU_GROUP_ID(64).
This can cause array outbound issue.

Hence, fix issue by iterating while assigning value till
MAX_MU_GROUP_ID(64)-1.

patch-dependency: 504-ath12k-add-debugfs-support.patch

Signed-off-by: Sarika Sharma <quic_sarishar@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/debugfs_sta.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/debugfs_sta.c b/drivers/net/wireless/ath/ath12k/debugfs_sta.c
index 22c8ac3..93d063c 100644
--- a/drivers/net/wireless/ath/ath12k/debugfs_sta.c
+++ b/drivers/net/wireless/ath/ath12k/debugfs_sta.c
@@ -181,7 +181,7 @@ void ath12k_debugfs_sta_add_tx_stats(struct ath12k_link_sta *arsta,
 	tx_stats->ru_start = peer_stats->ru_start;
 	tx_stats->ru_tones = peer_stats->ru_tones;
 
-	if (peer_stats->mu_grpid <= MAX_MU_GROUP_ID &&
+	if (peer_stats->mu_grpid < MAX_MU_GROUP_ID &&
 	    peer_stats->ppdu_type != HTT_PPDU_STATS_PPDU_TYPE_SU) {
 		if (peer_stats->mu_grpid & (MAX_MU_GROUP_ID - 1))
 			tx_stats->mu_group[peer_stats->mu_grpid] =
-- 
2.34.1

